### ISSUE TITLE
[java] False warning "Discontinue using Rule name …"

### ISSUE THREAD

**Issue by christianhujer** [pmd/pmd#3316]
**Affects PMD Version:** 
I'm using PMD version 6.35.0.

**Description:**
When using PMD, it prints warning messages about the discontinuation of rules.
I am not referencing these rules anywhere.
These rules must be coming from the built-in configuration of PMD.
This means that PMD creates warnings about its own built-in configuration.

Here is the warning output:
```
[WARNING] Discontinue using Rule name category/java/bestpractices.xml/UnusedImports as it is scheduled for removal from PMD. PMD 7.0.0 will remove support for this Rule.
[WARNING] Discontinue using Rule name category/java/codestyle.xml/DontImportJavaLang as it is scheduled for removal from PMD. PMD 7.0.0 will remove support for this Rule.
[WARNING] Discontinue using Rule name category/java/codestyle.xml/DuplicateImports as it is scheduled for removal from PMD. PMD 7.0.0 will remove support for this Rule.
[WARNING] Discontinue using Rule name category/java/errorprone.xml/ImportFromSamePackage as it is scheduled for removal from PMD. PMD 7.0.0 will remove support for this Rule.
```

I think it is reasonable to expect that the built-in rules like `category/java/bestpractices.xml`, `category/java/codestyle.xml`, and `category/java/errorprone.xml` are consistent with the implementation of PMD and thus using these files as-is should not produce any warnings.

**Exception Stacktrace:**
(No exception)

**Code Sample demonstrating the issue:**

You can try these projects to see the warnings:
* https://github.com/nelkinda/gameoflife-java
* https://github.com/nelkinda/gameoflife-kotlin

**Steps to reproduce:**
Please provide detailed steps for how we can reproduce the bug.

1.  Setup a Java or Kotlin project with Maven or Gradle that uses PMD 6.35.0
2.  Run PMD

**Running PMD through:** *Maven* and *Gradle*

Probably not relevant:
* Kubuntu 20.04.2 LTS
* OpenJDK 16 (managed by SDKMAN!)
* Maven 3.6.3, managed by Maven Wrapper
* Gradle 7.0.2, managed by Gradle Wrapper

The severity of this issue is low, it is merely an inconvenience.

**Comment by oowekyala** [comment]
Thanks for the report. This is indeed a bug, the intended behavior being [documented here](https://github.com/pmd/pmd/pull/2086/files#diff-bf15c19c9aade5bcd3f111bd6b5d3607467afc322f6d1b1ca8f4094dc28d2c98R93-R100)

**Comment by adangel** [comment]
I had a look at this - it's not reproducible with plain PMD (e.g. via command line). With your test project (gameoflife-java) I could only see the warnings when using maven and not gradle.

The maven warnings are created, because you use the default configuration provided by the maven-pmd-plugin. In version 3.14.0, the deprecated rules are still listed there (https://github.com/apache/maven-pmd-plugin/blob/maven-pmd-plugin-3.14.0/src/main/resources/rulesets/java/maven-pmd-plugin-default.xml) - but will be gone with the next release (https://github.com/apache/maven-pmd-plugin/blob/master/src/main/resources/rulesets/java/maven-pmd-plugin-default.xml).

So, in order to get rid of these warnings for your project, you need to configure the maven-pmd-plugin to use your own ruleset, e.g.

```xml
<configuration>
    <rulesets>
        <ruleset>config/pmd/pmdMain.xml</ruleset>
    </rulesets>
</configuration>
```

Since you want to use for testing a different ruleset, this stackoverflow question might be interesting for you: https://stackoverflow.com/questions/67845980/pmd-maven-plugin-how-to-specify-a-different-ruleset-for-tests

I've tested gameoflife-java with gradle using `./gradlew build`, but I couldn't see the PMD warnings about the deprecated rules. Can you confirm, that this issue was only maven-pmd-plugin related?

----

For completeness, here's the complete maven-pmd-plugin configuration for applying a different ruleset for main/test and also failing the build (note: pmd:pmd just creates the report, but pmd:check actually will fail the build):

```xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>3.14.0</version>
                <dependencies>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-core</artifactId>
                        <version>${version.pmd}</version>
                    </dependency>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-java</artifactId>
                        <version>${version.pmd}</version>
                    </dependency>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-javascript</artifactId>
                        <version>${version.pmd}</version>
                    </dependency>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-jsp</artifactId>
                        <version>${version.pmd}</version>
                    </dependency>
                </dependencies>
                <configuration>
                    <includeTests>false</includeTests>
                    <rulesets>
                        <ruleset>config/pmd/pmdMain.xml</ruleset>
                    </rulesets>
                    <linkXRef>false</linkXRef>
                </configuration>
                <executions>
                    <execution>
                        <id>pmd-main</id>
                        <phase>verify</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>pmd-test</id>
                        <phase>verify</phase>
                        <goals>
                            <goal>pmd</goal>
                            <goal>check</goal>
                        </goals>
                        <configuration>
                            <targetDirectory>${project.build.directory}/pmdTest/</targetDirectory>
                            <includeTests>true</includeTests>
                            <excludeRoots>
                                <excludeRoot>${basedir}/src/main/java</excludeRoot>
                            </excludeRoots>
                            <rulesets>
                                <ruleset>config/pmd/pmdTest.xml</ruleset>
                            </rulesets>
                        </configuration>
                    </execution>
                    <execution>
                        <id>cpd</id>
                        <phase>verify</phase>
                        <goals>
                            <goal>cpd-check</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

It has one downside though: PMD is run three times.... twice for the main code, and once for the test code. The reason is, that "pmd:check" triggers automatically "pmd:pmd", but it uses only the standard configuration. The executing "pmd-test" calls once "pmd:pmd" for the test code (which creates target/pmdTest/pmd.xml), the calls pmd:check - which itself calls pmd:pmd and uses the default configuration - and then runs pmd:check actually, which uses the execution configuration and uses target/pmdTest/pmd.xml to decide whether to fail the build.
